modules/common/src/encoding.ts

```diff
@@ -102,6 +102,35 @@ export function decodeRatio(
   return Number(onChainValue) / (10 ** digits);
 }
 
+/** Encodes the particles of a proposalId into a globally unque value for subgraph */
+export function encodeProposalId(pluginAddress: string, id: number) {
+  if (!/^0x[A-Fa-f0-9]{40}$/.test(pluginAddress)) {
+    throw new Error("Invalid address");
+  }
+
+  return `${pluginAddress}_0x${id.toString(16)}`;
+}
+
+/** Decodes a proposalId from subgraph and returns the original pluginAddress and the nonce */
+export function decodeProposalId(
+  proposalId: string,
+): { pluginAddress: string; id: number } {
+  if (!/^0x[A-Fa-f0-9]{40}_(0x[A-Fa-f0-9]{1,64})$/.test(proposalId)) {
```

Comment:
Using isProposalId() here now

Not super important

Reply:
done

Reply:
done